core-plugin/src/com/google/gct/idea/debugger/CloudDebuggerClient.java

```diff
@@ -105,15 +110,27 @@ public void statusChanged(boolean login) {
           HttpRequestInitializer initializer = new HttpRequestInitializer() {
             @Override
             public void initialize(HttpRequest httpRequest) throws IOException {
+              HttpHeaders headers = new HttpHeaders();
               httpRequest.setConnectTimeout(timeout);
               httpRequest.setReadTimeout(timeout);
+              httpRequest.setHeaders(headers);
               credential.initialize(httpRequest);
             }
           };
 
           HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
-          cloudDebuggerClient = new Builder(httpTransport, JSON_FACTORY, initializer).setRootUrl(ROOT_URL)
-            .setApplicationName(GoogleLoginUtils.getCurrentPlatformName()).build().debugger();
+          String userAgent = USER_AGENT;
+          try {
+            userAgent += GoogleLoginUtils.getCurrentPlatformName()
```

Comment:
You can check the android repo but I would hope that people are not using
this util class. Only the login specific API is expected to be in use.

On Fri, Jan 8, 2016 at 12:18 PM, Elliotte Rusty Harold <
notifications@github.com> wrote:

> In core-plugin/src/com/google/gct/idea/debugger/CloudDebuggerClient.java
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/327#discussion_r49212718>
> :
>
> >                credential.initialize(httpRequest);
> >              }
> >            };
> >
> >            HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
> > -          cloudDebuggerClient = new Builder(httpTransport, JSON_FACTORY, initializer).setRootUrl(ROOT_URL)
> > -            .setApplicationName(GoogleLoginUtils.getCurrentPlatformName()).build().debugger();
> > +          String userAgent = USER_AGENT;
> > +          try {
> > +            userAgent += GoogleLoginUtils.getCurrentPlatformName()
>
> Maybe. Let me polay wiht that and see. Who else besides us depends on the
> login plugin if I move things? (I notice there's a lot of code in there we
> don't use. I assume someone does.)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/327/files#r49212718>
> .
>